docs/InspectingContracts.md

```diff
@@ -0,0 +1,140 @@
+        "transmitter": "terra1myd0kxk3fqaz9zl47gm2uvxjm0zn3lczrtvljz"
+    }
+}
+```
+
+You can also perform string queries if you use escape characters.
+
+```json
+"\"owner\""
+```
+
+### Queries Via Terrad
+
+Terrad is a CLI tool that enables users to interact with the Terra blockchain. To query a contract, you must use *terrad query wasm contract-store*.
+
+```bash
+Usage:
+  terrad query wasm contract-store [bech32-address] [msg] [flags]
+
+Flags:
+      --height int      Use a specific height to query state at (this can error if the node is pruning state)
+  -h, --help            help for contract-store
+      --node string     <host>:<port> to Tendermint RPC interface for this chain (default "tcp://localhost:26657")
+  -o, --output string   Output format (text|json) (default "text")
+
+Global Flags:
+      --chain-id string     The network chain ID
+      --home string         directory for config and data (default "/Users/kylemartin/.terra")
+      --log_format string   The logging format (json|plain) (default "plain")
+      --log_level string    The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
+      --trace               print out full stack trace on errors
+```
+
+Below is an example query to the OCR2 contract. 
+
+```bash
+terrad query wasm contract-store terra14mf0qcjpduhcs8p6289pjnwn8skhgk5aus3yxg '{"owed_payment":{"transmitter": "terra1myd0kxk3fqaz9zl47gm2uvxjm0zn3lczrtvljz"}}' --node "https://RPC_URL:443"
+```
+
+The response is as follows.
+
+```bash
+query_result: "3013470097000000000"
+```
+
+### Queries Via cURL
+
+You can also interact with deployed contracts via cURL. Queries using cURL will have the following format, where *query* is either a string or a URL encoded JSON object.
+
+```bash
+curl '${fcd_endpoint}/wasm/contracts/${address}/store?query_msg=${query}'
+```
+
+Some examples of how to query the USD/USDT proxy contract can be found below.
+
+```bash
+curl 'https://bombay-fcd.terra.dev/wasm/contracts/terra134lgzqfwms0sg4a33wpygj8waff2d704gcezsu/store?query_msg="owner"'
+curl 'https://bombay-fcd.terra.dev/wasm/contracts/terra134lgzqfwms0sg4a33wpygj8waff2d704gcezsu/store?query_msg=%7B%22round_data%22:%7B%22round_id%22:4294968601%7D%7D'
+```
+
+Note that *%7B%22round_data%22:%7B%22round_id%22:4294968601%7D%7D* is the URL encoded format of *{"round_data":{"round_id":4294968601}}*. The sample responses for each of the cURL commands above are as follows.
+
+```bash
+# curl 'https://bombay-fcd.terra.dev/wasm/contracts/terra134lgzqfwms0sg4a33wpygj8waff2d704gcezsu/store?query_msg="owner"'
+{"height":"8260930","result":"terra19mz966zzv34tr7vxu0z66ps2ey20mv3nfdzukd"}
+# curl 'https://bombay-fcd.terra.dev/wasm/contracts/terra134lgzqfwms0sg4a33wpygj8waff2d704gcezsu/store?query_msg=%7B%22round_data%22:%7B%22round_id%22:4294968601%7D%7D'
+{"height":"8260933","result":{"round_id":4294968601,"answer":"521361092112","observations_timestamp":1646914654,"transmission_timestamp":1646914668}}
+```
```

Comment:
In the near future, we should add a `## Queries Via Gauntlet` section and make Gauntlet capable of all of this and more with typed contract inspections (`ocr2:inspect`, etc.).

We could have a general command like: `yarn gauntlet contract:query --network [NETWORK] --msg '{"round_data":{"round_id":4294968601}}' [ADDR]`

Something to keep in mind as the next steps to improve Gauntlet API, usage, and docs. - cc @RodrigoAD

Reply:
Extracted this as an GH Issue:

- https://github.com/smartcontractkit/chainlink-terra/issues/227